src/libraries/System.Text.Json/tests/Common/ConstructorTests/ConstructorTests.ParameterMatching.cs

```diff
@@ -1322,5 +1322,49 @@ public class ClassWithIgnoredSameType
 
             public ClassWithIgnoredSameType(ClassWithIgnoredSameType prop) { }
         }
+
+        [Fact]
+        public void StructWithPropertyInit_DeseralizeEmptyObject()
+        {
+            string json = @"{}";
+            var obj = JsonSerializer.Deserialize<StructWithPropertyInit>(json);
+            Assert.Equal(42, obj.A);
+            Assert.Equal(0, obj.B);
+        }
+
+        public struct StructWithPropertyInit
+        {
+            public long A { get; set; } = 42;
```

Comment:
consider for completeness one more test where you explicitly set this to something else (i.e. 43) in the JSON and make sure this got respected

Reply:
Added!